brave/brave-core
# Implement eip-6963

Resolves https://github.com/brave/brave-browser/issues/30595


Resolves 



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1) Set default ethereum wallet to Brave Wallet
2) Check that window.braveEthereum and window.ethereum are created and isBraveWallet both
3) Test that https://eip6963.org/ sees Brave Wallet
4) Install Metamask
5) Check that window.ethereum is not overwritten by MetaMask and check that https://eip6963.org/ sees Brave Wallet
----
1) Set default ethereum wallet to Extensions(Brave Wallet fallback)
2) Check that window.braveEthereum and window.ethereum are created
3) Test https://eip6963.org/
4) Install Metamask
5) Check that window.ethereum is overwritten by Metamask Extension while window.braveEthereum is still isBraveWallet and https://eip6963.org/ sees Brave Wallet
----
1) Set default ethereum wallet to None
2) Check that there is no window.braveEthereum and window.ethereum
3) Check that https://eip6963.org/ doesn't see Brave Wallet
